client/src/components/Home.jsx

```diff
@@ -138,11 +149,24 @@ const Home = ({ setIsLoading, isMobile, isLoggedIn, isLoading, setIsLoggedIn, sh
   }
 
   const addCategory = (category) => {
-    let incomingId = info.calendars[0].calendar_id;
+
+    let incomingId;
+
+    if (info.length > 0) {
+      incomingId = info.calendars[0].calendar_id;
+    } else {
+      incomingId = 11;
+    }
 
     axios.post('http://localhost:3000/todoList/category', {  params: { calendar_id: incomingId, category: category} })
       .then((result) => {
-        console.log(result);
+        console.log('cat post result: ', result);
+        let catId = result.data.category_id;
+        let newCat = {category_id: catId, category: category, todoitems: []};
+        let newEventsList = myEvents[0];
+        newEventsList.push(newCat);
+        setMyEvents(newEventsList);
```

Comment:
Is this working for you ? Do you have to get in one-level deeper of nesting ? 
myEvents[0] --> object --> myEvents[0].categories --> array of objects